enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,235 @@
+A user must have explicit RBAC rules, `managedclustersets/join`, to add such labels.
+There are also raising requirements on using managedClusterSet for the following purposes. Within a purpose, the managedClusterSet must be exclusive, but for different purposes it can overlap.
+
+- As cluster admin, I want to create a managedClusterSet which includes all clusters in the apac region. Then I can apply certain configurations to all clusters in apac.
+- As a SRE team member, I want to use a managedClusterSet for Disaster Recovery purposes. The managedClusterSet should select two or more clusters in different regions. 
+- As a DEV team member, I want to use a managedClusterSet to select clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+
+## Proposal
+
+### Changes on managedClusterSet APIs.
+
+Currently, managedClusterSets use the label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` to select managedClusters. 
+But for some use cases, I want to use other labels to select managedClusters in a managedClusterSets. Meanwhile, managedClusterSets should be exclusive within a purpose, but for different purposes it can overlap.
+So, In this proposal, managedClusterSets could use any labels to select managedClusters. 
+
+So, we add a field `ClusterLabel` into managedClusterSet spec
+
+```go
+ type ManagedClusterSetSpec struct {
+ // ClusterLabel defines the label which the managedClusterSet will use to select managedClusters.
+ // If the field is not set, we will use "cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>" as default value.
+ // +optional
+ ClusterLabel ClusterLabel  `json:"clusterLabel"`
+}
+
+//ClusterLabel defines one cluster label
+type ClusterLabel struct {
+ Key string `json:"key"`
+ Value string `json:"value"`
+}
+```
+
+### RBAC
+Currently, When I want to create a managedClusterSet, I only should have `create` permission to the managedClusterSet. 
+In the managedCluster part, label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` means the cluster is in a managedClusterSet.
+If I want to add the label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` to a managedCluster, I must have `create` permission for subresource `managedclustersets/join`. 
+If I want to add/update other labels, I just need to have `update` permission to the managedCluster.
+
+In this proposal, managedClusterSet could use any labels to select managedClusters. And if I add a label for a managedCluster, it may lead to the managedCluster added to a managedClusterSet. So we need to rethink about the rbac control of managedClusterSet
+
+We will use the following questions to consider the RBAC control.
+
+#### What kinds of roles exist related to managedClusterSet
+In General, there are three kinds of roles related to managedClusterSet.
+- Cluster admin 
```

Comment:
can you elaborate what actions each role will make here?

Reply:
done